packages/frontend/src/pages/Users/components/UsersTable.tsx

```diff
@@ -1,121 +1,159 @@
+        setTableData(allQuantifierUsers);
+        break;
+    }
+    setFilter('');
+  }, [selectedRole]);
 
-  const handleDeleteQuantifierClick = (quantifier: UserDto): void => {
-    setSelectedQuantifier(quantifier);
-    setIsOpen(true);
-  };
+  React.useEffect(() => {
+    if (tableData) {
+      setPage(1);
 
-  const removeQuantifier = (id: string): void => {
-    void removeRole(id, UserRole.QUANTIFIER);
-  };
+      if (tableData.length % 5 === 0) {
+        setLastPage(Math.trunc(tableData.length / 5));
+      } else {
+        setLastPage(Math.trunc(tableData.length / 5) + 1);
+      }
+    }
+  }, [tableData]);
 
   return (
-    <table className="w-full table-auto" {...getTableProps()}>
-      <tbody {...getTableBodyProps()}>
-        {rows.map((row) => {
-          prepareRow(row);
-          return (
-            // eslint-disable-next-line react/jsx-key
-            <tr {...row.getRowProps()}>
-              {row.cells.map((cell) => {
-                // eslint-disable-next-line react/jsx-key
-                return <td {...cell.getCellProps()}>{cell.render('Cell')}</td>;
-              })}
-              <td className="w-8">
-                <button
-                  onClick={(): void =>
-                    handleDeleteQuantifierClick(row.original as UserDto)
-                  }
-                  className="hover:text-red-600"
-                >
-                  <FontAwesomeIcon
-                    icon={faTimesCircle}
-                    size="1x"
-                    className="inline-block"
-                  />
-                </button>
-              </td>
-            </tr>
-          );
-        })}
-
-        {isOpen ? (
-          <Dialog
-            open={isOpen && !!selectedQuantifier}
-            onClose={(): void => setIsOpen(false)}
-            className="fixed inset-0 z-10 overflow-y-auto"
-            initialFocus={deleteDialogRef}
+    <div>
+      <div className="flex gap-8">
+        <select
+          className="bg-transparent border-black"
+          onChange={(event: React.ChangeEvent<HTMLSelectElement>): void =>
+            setSelectedRole(UserRole[event.target.value])
+          }
+        >
+          {roleOptions.map((option) => (
+            <option key={option.name} value={option.value}>
+              {option.name}
+            </option>
+          ))}
+        </select>
+        <div className="flex items-center border border-black">
+          <label className="relative">
+            <FontAwesomeIcon
+              icon={faMagnifyingGlass}
+              size="1x"
+              className="absolute top-1/2 transform -translate-y-1/2 left-3"
+            />
+            <input
+              type="text"
+              name="search"
+              placeholder="Search"
+              className="bg-transparent outline-none border-none focus:ring-0 block pl-8"
+              value={filter}
+              onChange={(event: React.ChangeEvent<HTMLInputElement>): void =>
```

Comment:
The search results update when I *add* characters to the search text, but not when I *remove* characters.

It also needs to reset to showing all results when the search query is empty.